src/Collections/CachedConfigCollection.php

```diff
@@ -27,6 +27,11 @@ class CachedConfigCollection implements ConfigCollectionInterface
      */
     protected $collection;
 
+    /**
+     * Stores a hash to allow comparison
+     */
+    protected ?string $collectionHash = null;
```

Comment:
```suggestion
    private ?string $collectionHash = null;
```
`protected` is new API while `private` isn't. This makes it easier to maintain. I can't think of any reason for a subclass to need to access $collectionHash. If it turns out there is a reason, we can change it to `protected` later on